tool/src/main.rs

```diff
@@ -54,12 +58,24 @@ fn main() {
         tas = Tas::new().unwrap();
         println!("TAS created successfully.");
     }
-
-    let script_file = env::args().skip(1).next().unwrap_or("tas.lua".to_string());
+    let script_file = if env::args().len() == 1 {
+        if std::env::current_dir().unwrap().join("tas.lua").metadata().is_err() {
+            let cur_dir = std::env::current_dir().unwrap();
+            let cur_dir = cur_dir.to_str().unwrap();
+            let result = nfd::open_file_dialog(Some("lua"), Some(cur_dir)).unwrap_or_else(|e| { panic!(e) });
+            match result {
+                Response::Okay(file_path) => Path::new(&file_path).to_str().unwrap().to_string(),
```

Comment:
You already have a path, but you convert it to a string. The method `Tas::execute` takes `AsRef<Path>`, which is implemented by a multitude of types, including `Path`, `PathBuf`, `str` and `String`. So you can pass any of those to the function.
In this case, all return values of the if-statement need to be the same, because they are assigned to the same variable.
The advantage of Path / PathBuf is that str / String require their content to be UTF-8, but sometimes paths aren't UTF-8. Those cases are handled correctly by Path / PathBuf, which are therefore better to use for handling paths / files than str / String.
You need to return a value from the if-statement, which requires the value to be owning, which leaves us with `PathBuf`.

So please change all paths to return a `PathBuf` instead of `String`.